velox/exec/tests/TableWriteTest.cpp

```diff
@@ -610,3 +644,82 @@ TEST_F(TableWriteTest, writeNoFile) {
   execute(plan, std::make_shared<core::QueryCtx>(executor_.get()));
   ASSERT_TRUE(fs::is_empty(outputDirectory->path));
 }
+
+TEST_F(TableWriteTest, errorWhenImmutablePartitionsEnabled) {
+  auto tableType = connector::hive::LocationHandle::TableType::kExisting;
+  auto vectors = makeVectors(rowType_, 10, 10);
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = createInsertPlan(rowType_, vectors, tableType, outputDirectory);
+
+  VELOX_ASSERT_THROW(
+      AssertQueryBuilder(plan)
+          .connectorConfig(
+              kHiveConnectorId,
+              HiveConfig::kImmutablePartitions,
+              folly::to<std::string>("true"))
+          .copyResults(pool()),
+      "Unpartitioned Hive tables are immutable.");
+}
+
+TEST_F(TableWriteTest, immutablePartitionsInsertZeroRows) {
```

Comment:
Its a bit odd to call this test "immutablePartitionsInsertZeroRows" test when the partitions are mutable. Maybe just call it insertZeroRows.